File: leetcode/algorithms/go/0001-two-sum.go
package leetcode

func twoSum(nums []int, target int) []int {
	var m = make(map[int]int, len(nums))
	for i, num := range nums {
		if j, ok := m[target-num]; ok {
			return []int{j, i}
		}
		m[num] = i
	}
	return nil
}

// func twoSum(nums []int, target int) []int {
// 	var m = make(map[int]int, len(nums))
// 	for i, num := range nums {
// 		if j, ok := m[target-num]; ok && j != i {
// 			return []int{j, i}
// 		}
// 		m[num] = i
// 	}
// 	return nil
// }

File: leetcode/algorithms/javascript/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let m = new Map();
    for (let i = 0; i < nums.length; i++) {
        let j = m.get(target - nums[i]);
        if (j != undefined && j != i) {
            return [j, i];
        }
        m.set(nums[i], i);
    }
};
